shell.nix

```diff
@@ -17,5 +17,53 @@ let
   pkgs = import nixpkgsSrc { inherit system; };
   devshell = import devshellSrc { inherit system pkgs; };
 
+  withCategory = category: attrset: attrset // { inherit category; };
+  util = withCategory "utils";
+
+  test = name: withCategory "tests" {
+    name = "check-${name}";
+    help = "Checks ${name} testcases";
+    command = "cd $DEVSHELL_ROOT/tests/${name} && nix flake show && nix flake check";
+  };
+
+  dry-nixos-build = example: host: withCategory "dry-build" {
+    name = "build-${example}-${host}";
+    command = "cd $DEVSHELL_ROOT/examples/${example} && nix flake show && nix build .#nixosConfigurations.${host}.config.system.build.toplevel --dry-run";
+  };
```

Comment:
I don't know if this would be necessary but the commands could be prepended with `nix flake lock --update-input utils`